flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultExecutionDeployerTest.java

```diff
@@ -0,0 +1,388 @@
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+import org.apache.flink.runtime.shuffle.TestingShuffleMaster;
+import org.apache.flink.runtime.testtasks.NoOpInvokable;
+import org.apache.flink.util.ExecutorUtils;
+import org.apache.flink.util.IterableUtils;
+import org.apache.flink.util.TestLoggerExtension;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** Tests for {@link DefaultExecutionDeployer}. */
+@ExtendWith(TestLoggerExtension.class)
+public class DefaultExecutionDeployerTest {
+
+    private ScheduledExecutorService executor;
+    private ComponentMainThreadExecutor mainThreadExecutor;
+    private TestExecutionOperationsDecorator testExecutionOperations;
+    private ExecutionVertexVersioner executionVertexVersioner;
+    private TestExecutionSlotAllocator testExecutionSlotAllocator;
+    private TestingShuffleMaster shuffleMaster;
+    private TestingJobMasterPartitionTracker partitionTracker;
+    private Time partitionRegistrationTimeout;
+
+    @BeforeEach
+    void setUp() {
+        executor = Executors.newSingleThreadScheduledExecutor();
+        mainThreadExecutor = ComponentMainThreadExecutorServiceAdapter.forMainThread();
+        testExecutionOperations =
+                new TestExecutionOperationsDecorator(
+                        new ExecutionOperations() {
+                            @Override
+                            public void deploy(Execution execution) {}
+
+                            @Override
+                            public CompletableFuture<?> cancel(Execution execution) {
+                                return null;
+                            }
+
+                            @Override
+                            public void markFailed(Execution execution, Throwable cause) {}
+                        });
+        executionVertexVersioner = new ExecutionVertexVersioner();
+        testExecutionSlotAllocator = new TestExecutionSlotAllocator();
+        shuffleMaster = new TestingShuffleMaster();
+        partitionTracker = new TestingJobMasterPartitionTracker();
+        partitionRegistrationTimeout = Time.milliseconds(5000);
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (executor != null) {
+            ExecutorUtils.gracefulShutdown(5, TimeUnit.SECONDS, executor);
+        }
+    }
+
+    @Test
+    public void testDeployTasks() throws Exception {
```

Comment:
```suggestion
    void testDeployTasks() throws Exception {
```

Reply:
All test cases are best not to be public.

Reply:
Ok.